atlasdb-api/src/main/java/com/palantir/atlasdb/spi/SharedResourcesConfig.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.spi;
+
+import org.immutables.value.Value;
+
+@Value.Immutable
+public interface SharedResourcesConfig {
+    int sharedGetRangesPoolSize();
+
+    int sharedKvsExecutorSize();
+
+    LocalConnectionConfig connectionConfig();
```

Comment:
Is this common to all KVSes? Feels a bit like a DbKVS specific thing: I'm fine with us not doing the specific sub-impls right now, but we should probably leave a note here that this is used for relational key-value-services

Reply:
It actually can be. As it's currently wired, Cassandra KVS will actually wire the get ranges executors to be shared if this is set, but there is no reason we can't in the future wire up other ones if that is useful